src/Compilers/CSharp/Test/Syntax/Parsing/MemberDeclarationParsingTests.cs

```diff
@@ -906,6 +911,608 @@ public void SetAndInitAccessor()
+            N(SyntaxKind.PropertyDeclaration);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "required");
+                }
+                N(SyntaxKind.IdentifierToken, "Prop");
+                N(SyntaxKind.AccessorList);
+                {
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.GetAccessorDeclaration);
+                    {
+                        N(SyntaxKind.GetKeyword);
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+            }
+            EOF();
+        }
+
+        [Fact, CompilerTrait(CompilerFeature.RequiredMembers)]
+        public void RequiredModifierProperty_03()
+        {
+            UsingDeclaration("required Prop { get; }", options: RequiredMembersOptions,
+                // (1,1): error CS1073: Unexpected token '{'
+                // required Prop { get; }
+                Diagnostic(ErrorCode.ERR_UnexpectedToken, "required Prop").WithArguments("{").WithLocation(1, 1),
+                // (1,15): error CS1519: Invalid token '{' in class, record, struct, or interface member declaration
+                // required Prop { get; }
+                Diagnostic(ErrorCode.ERR_InvalidMemberDecl, "{").WithArguments("{").WithLocation(1, 15)
+                );
+            N(SyntaxKind.IncompleteMember);
+            {
+                N(SyntaxKind.RequiredKeyword);
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "Prop");
+                }
+            }
+            EOF();
+        }
+
+        [Theory, CompilerTrait(CompilerFeature.RequiredMembers)]
+        [MemberData(nameof(ParseOptionsData))]
+        public void RequiredModifierProperty_04(CSharpParseOptions parseOptions)
+        {
+            UsingDeclaration("required required { get; }", options: parseOptions);
+            N(SyntaxKind.PropertyDeclaration);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "required");
+                }
+                N(SyntaxKind.IdentifierToken, "required");
+                N(SyntaxKind.AccessorList);
+                {
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.GetAccessorDeclaration);
+                    {
+                        N(SyntaxKind.GetKeyword);
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+            }
+            EOF();
+        }
+
+        [Fact, CompilerTrait(CompilerFeature.RequiredMembers)]
+        public void RequiredModifierProperty_05()
+        {
+            UsingDeclaration("required required { get; }", options: RequiredMembersOptions,
```

Comment:
I assume we parsed as duplicate `required` modifier. Is that correct?

Reply:
Yep. That's why there's two `RequiredKeyword`s in the incomplete member below.